### PR TITLE
KAFKA-13445: Add ECDSA test for JWT validation

### PR DESCRIPTION
**Pull request by kirktrue**
The tests for OAuth JWT validation all assume usage of RSA, but we need to have ECDSA support there too.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/common/security/oauthbearer/secured/AccessTokenBuilder.java [hunk 1]
```diff
@@ -49,22 +49,24 @@
...
 
-    public static RsaJsonWebKey createJwk() throws JoseException {
-        RsaJsonWebKey jwk = RsaJwkGenerator.generateJwk(2048);
-        jwk.setKeyId("key-1");
-        return jwk;
+    public String alg() {
+        return alg;
+    }
+
+    public AccessTokenBuilder alg(String alg) {
```
Comment (junrao) [inline clients/src/test/java/org/apache/kafka/common/security/oauthbearer/secured/AccessTokenBuilder.java:67]: While you are at this, a few other methods like audience() and subjectClaimName seem never used. Is that expected?

File: clients/src/test/java/org/apache/kafka/common/security/oauthbearer/secured/ValidatorAccessTokenValidatorTest.java [hunk 2]
```diff
@@ -38,14 +40,35 @@ protected AccessTokenValidator createAccessTokenValidator(AccessTokenBuilder bui
...
+    private void testEncryptionAlgorithm(PublicJsonWebKey jwk, String alg) throws Exception {
+        AccessTokenBuilder builder = new AccessTokenBuilder().jwk(jwk).alg(alg);
         AccessTokenValidator validator = createAccessTokenValidator(builder);
 
         JsonWebSignature jws = new JsonWebSignature();
         jws.setKey(builder.jwk().getPrivateKey());
         jws.setKeyIdHeaderValue(builder.jwk().getKeyId());
-        jws.setAlgorithmHeaderValue(AlgorithmIdentifiers.RSA_USING_SHA256);
+        jws.setAlgorithmHeaderValue(alg);
+        jws.setPayload("{}");
```
Comment (junrao) [inline clients/src/test/java/org/apache/kafka/common/security/oauthbearer/secured/ValidatorAccessTokenValidatorTest.java]: Hmm, how is jws being used?

**Review by junrao — COMMENTED** [review]
@kirktrue : Thanks for the PR. A couple of comments below.

File: clients/src/test/java/org/apache/kafka/common/security/oauthbearer/secured/ValidatorAccessTokenValidatorTest.java (hunk 2 above)
Comment (kirktrue) [reply clients/src/test/java/org/apache/kafka/common/security/oauthbearer/secured/ValidatorAccessTokenValidatorTest.java]: It isn't. I've removed it.

File: clients/src/test/java/org/apache/kafka/common/security/oauthbearer/secured/AccessTokenBuilder.java (hunk 1 above)
Comment (kirktrue) [reply clients/src/test/java/org/apache/kafka/common/security/oauthbearer/secured/AccessTokenBuilder.java:67]: They're there for completeness. I've removed them now.

**Review by junrao — APPROVED** [review]
@kirktrue : Thanks for the PR. LGTM

**Pull request merged**